utils/src/utils.cpp

```diff
@@ -35,6 +37,29 @@ std::string hex_to_bytes(const std::string &hex) {
     return result;
 }
 
+namespace base64 = boost::beast::detail::base64;
+
+// base64 stuff was copied from boost 1.66 sources
+std::string base64_decode(std::string const& data) {
+    std::string dest;
+    dest.resize(base64::decoded_size(data.size()));
+    auto const result = base64::decode(&dest[0], data.data(), data.size());
+    dest.resize(result.first);
+    return dest;
+}
+
+static std::string base64_encode(std::uint8_t const* data, std::size_t len) {
+    std::string dest;
+    dest.resize(base64::encoded_size(len));
+    dest.resize(base64::encode(&dest[0], data, len));
+    return dest;
+}
+
+std::string base64_encode(std::string const& s) {
+    return base64_encode(reinterpret_cast<std::uint8_t const*>(s.data()),
+                         s.size());
+}
```

Comment:
Won't this still have same problem if they move the stuff out from detail namespace again?

I'd just copy the concrete implementation or take it from lokinet assuming they have one. 
Otherwise gnulib have one, where we took our exp2/round from in Loki Daemon with a permissive license.

https://github.com/coreutils/gnulib/blob/master/lib/base64.h

Reply:
Sure, I just wanted it to compile. Added a todo to use a different implementation.